admin/class-taxonomy-presenter.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Class WPSEO_Taxonomy_Presenter
+ */
+class WPSEO_Taxonomy_Presenter {
+
+	/**
+	 * @var array
+	 */
+	private $options = array();
+
+	/**
+	 * @var bool|mixed
+	 */
+	private $tax_meta;
+
+	/**
+	 * @var stdClass
+	 */
+	private $term;
+
+	/**
+	 * @var array   Options array for the no-index options, including translated labels
+	 */
+	private $no_index_options = array();
+
+	/**
+	 * @var array   Options array for the sitemap_include options, including translated labels
+	 */
+	private $sitemap_include_options = array();
+
+	/**
+	 * @param stdClass $term
+	 */
+	public function __construct( $term ) {
+		$this->term     = $term;
+		$this->tax_meta = WPSEO_Taxonomy_Meta::get_term_meta( (int) $term->term_id, $term->taxonomy );
+		$this->options  = WPSEO_Options::get_all();
+
+		$this->translate_meta_options();
+	}
+
+	/**
+	 * Displaying the form fields
+	 *
+	 * @param array $fields
+	 */
+	public function display_fields( array $fields ) {
+		foreach ( $fields as $field_name => $field_options ) {
+			if ( empty( $field_options['hide'] ) ) {
+				$this->form_row( 'wpseo_' . $field_name, $field_options['label'], $field_options['description'], $field_options['type'], $field_options['options'] );
+			}
+		}
+	}
+
+	/**
+	 * Getting the fields for the general tab
+	 *
+	 * @return array
+	 */
+	public function general_fields() {
+
+		$current = 'index';
+		if ( isset( $options[ 'noindex-tax-' . $this->term->taxonomy ] ) && $this->options[ 'noindex-tax-' . $this->term->taxonomy ] === true ) {
```

Comment:
adding `$options[ 'noindex-tax' . $this->term->taxonomy` will increase readability. You're performing the same call twice to check if the variable is set and is true. By putting it in a single variable and then checking if the variable is set and true, the call won't have to be made twice.